kythe/go/platform/kzip/kzip.go

```diff
@@ -316,11 +392,18 @@ type Writer struct {
 	fd  stringset.Set // file digests already written
 	ud  stringset.Set // unit digests already written
 	c   io.Closer     // a closer for the underlying writer (may be nil)
+
+	encoding Encoding // What encoding to use
+}
+
+// WriterOptions describe options when creating a Writer
+type WriterOptions struct {
+	Encoding Encoding
```

Comment:
Typically, (although not always), a named function is used for options, e.g.

`type WriterOption func(*Writer)`
```
func NewWriter(w io.Writer, options ...WriterOption) (*Writer, error) {
  ...
  for _, opt := range options {
    opt(w)
  }
}

func WithEncoding(e Encoding) WriterOption {
  return func(w *Writer) {
    w.encoding = e
  }
}
```

Reply:
This also obviates the "WithOptions" constructor.